lib/path.js

```diff
@@ -29,6 +29,17 @@ function assertPath(path) {
   }
 }
 
+function findLastSlashIndex(path, isWin32) {
+  var i = path.length - 1;
```

Comment:
I think the logic here can be simplified to `Math.max(path.lastIndexOf(isWin32 ? '\\' : '/'), 0)`? Or does iterating over the string with `charCodeAt` yield any performance benefit?

Reply:
Here is just inherited from the logic before. I've just removed it and used lastIndexOf.